gdal/ogr/ogrsf_frmts/sqlite/ogrsqlitelayer.cpp

```diff
@@ -373,11 +373,17 @@ void OGRSQLiteLayer::BuildFeatureDefn( const char *pszLayerName,
             {
                 oField.SetType(OFTStringList);
             }
-            else if (EQUAL(pszDeclType, "BIGINT") || EQUAL(pszDeclType, "INT8"))
+            else if (EQUAL(pszDeclType, "BIGINT") || EQUAL(pszDeclType, "INT8") || 
+                     EQUAL(pszDeclType, "UNSIGNED BIG INT"))
             {
                 oField.SetType(OFTInteger64);
             }
-            else if (STARTS_WITH_CI(pszDeclType, "INTEGER"))
+            else if (STARTS_WITH_CI(pszDeclType, "INTEGER") ||
+                     EQUAL(pszDeclType, "INT") ||
+                     EQUAL(pszDeclType, "TINYINT") ||
+                     EQUAL(pszDeclType, "SMALLINT") ||
+                     EQUAL(pszDeclType, "MEDIUMINT") ||
+                     EQUAL(pszDeclType, "INT2"))
```

Comment:
as a small improvement, when INT2, oField.SetSubType(OFSTInt16) could be set to add a hint this is a short integer.